digdag-spi/src/main/java/io/digdag/spi/TaskExecutionException.java

```diff
@@ -37,40 +93,80 @@ private TaskExecutionException(int retryInterval, ConfigElement stateParams)
         this.stateParams = Optional.of(stateParams);
     }
 
+    private TaskExecutionException(Throwable cause, ConfigElement error, int retryInterval, ConfigElement stateParams)
+    {
+        super(cause);
+        this.error = Optional.of(error);
+        this.retryInterval = Optional.of(retryInterval);
+        this.stateParams = Optional.of(stateParams);
+    }
+
     private final Optional<ConfigElement> error;
     private final Optional<Integer> retryInterval;
     private final Optional<ConfigElement> stateParams;
 
-    public TaskExecutionException(Throwable cause, ConfigElement error)
+    /**
+     * Wrap an expected exception to make the task failed.
+     */
+    public TaskExecutionException(Throwable cause)
     {
-        super(cause);
-        this.error = Optional.of(error);
+        this(formatExceptionMessage(cause), cause);
+    }
+
+    /**
+     * Wrap an expected exception with a custom well-formatted message to make the task failed.
+     */
+    public TaskExecutionException(String customMessage, Throwable cause)
+    {
+        super(customMessage, cause);
+        this.error = Optional.of(buildExceptionErrorConfig(customMessage, cause));
         this.retryInterval = Optional.absent();
         this.stateParams = Optional.absent();
     }
 
-    public TaskExecutionException(String message, ConfigElement error)
+    /**
+     * Build an expected exception with a simple message to make the task failed.
+     */
+    public TaskExecutionException(String message)
+    {
+        this(message, ImmutableMap.of());
+    }
+
+    /**
+     * Build an expected exception with a simple message and properties to make the task failed.
+     */
+    public TaskExecutionException(String message, Map<String, String> errorProperties)
     {
         super(message);
-        this.error = Optional.of(error);
+        this.error = Optional.of(buildPropertiesErrorConfig(message, errorProperties));
         this.retryInterval = Optional.absent();
         this.stateParams = Optional.absent();
     }
 
-    public TaskExecutionException(Throwable cause, ConfigElement error, int retryInterval, ConfigElement stateParams)
+    private static ConfigElement buildPropertiesErrorConfig(String message, Map<String, String> errorProperties)
     {
-        super(cause);
-        this.error = Optional.of(error);
-        this.retryInterval = Optional.of(retryInterval);
-        this.stateParams = Optional.of(stateParams);
+        ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+        builder.putAll(errorProperties);
+        builder.put("message", message);
+        return ConfigElement.ofMap(builder.build());
     }
 
-    public TaskExecutionException(String message, ConfigElement error, int retryInterval, ConfigElement stateParams)
+    @Deprecated
+    public TaskExecutionException(String message, ConfigElement error)
```

Comment:
https://github.com/treasure-data/digdag/blob/c4ad2dcdabf3653c4bc3ef1babfdf098ecba2df5/digdag-standards/src/main/java/io/digdag/standards/operator/aws/EmrOperatorFactory.java#L398-L398
https://github.com/treasure-data/digdag/blob/c4ad2dcdabf3653c4bc3ef1babfdf098ecba2df5/digdag-standards/src/main/java/io/digdag/standards/operator/aws/EmrOperatorFactory.java#L522-L522

These codes pass ConfigElement.empty() to this method and then it seems `error` doesn't have `message` field. So we'd better take care of them?